### ISSUE TITLE
Add the ability to change password from edit profile form

### ISSUE THREAD

**Issue by mehreeee** [unstructuredstudio/zubhub#904]
There is no option for users to be able to change passwords. If the user needs to change the password, they should do it by entering old and new passwords… then the dev can perform a check. I suggest there should be another page or feature for users to be able to change passwords.
![8DD6EBED-5CD7-488C-B07C-B8DAD8C114F8](https://github.com/unstructuredstudio/zubhub/assets/105458062/fd28f760-5fa0-4f9f-bad0-8e23bc9092b2)

**Comment by Boluwatifey1** [comment]
Hi @mehreeee i discovered that too and i made a design for it but it was closed by @srish 

i added a change icon field beside the old password

![add Profile picture (6)](https://github.com/unstructuredstudio/zubhub/assets/127180289/044b9539-f574-4973-a895-014846be4941)

**Comment by aqsaaqeel** [comment]
> There is no option for users to be able to change passwords. If the user needs to change the password, they should do it by entering old and new passwords… then the dev can perform a check. I suggest there should be another page or feature for users to be able to change passwords. ![8DD6EBED-5CD7-488C-B07C-B8DAD8C114F8](https://user-images.githubusercontent.com/105458062/275194164-fd28f760-5fa0-4f9f-bad0-8e23bc9092b2.jpeg)

hey @mehreeee I think there used to be a `change password` feature but was removed, not sure about it you can check with mentors. But the basic idea behind not having a `change password` feature is that the average age of our users is very young. So, our main goal is to make it very simple for them to use. Right now, they can view their current password so that takes away the need to be able to change the password. Hope this makes sense!

cc: @srish @tuxology

**Comment by coderatomy** [comment]
Support for this is in place both backend and frontend. The only limitation is [here](https://github.com/unstructuredstudio/zubhub/issues/661#issuecomment-1470823228). I think this can't stop designers from working on the issue. Let's just make everything ready.

**Comment by mehreeee** [comment]
> > There is no option for users to be able to change passwords. If the user needs to change the password, they should do it by entering old and new passwords… then the dev can perform a check. I suggest there should be another page or feature for users to be able to change passwords. ![8DD6EBED-5CD7-488C-B07C-B8DAD8C114F8](https://user-images.githubusercontent.com/105458062/275194164-fd28f760-5fa0-4f9f-bad0-8e23bc9092b2.jpeg)
> 
> hey @mehreeee I think there used to be a `change password` feature but was removed, not sure about it you can check with mentors. But the basic idea behind not having a `change password` feature is that the average age of our users is very young. So, our main goal is to make it very simple for them to use. Right now, they can view their current password so that takes away the need to be able to change the password. Hope this makes sense!
> 
> cc: @srish @tuxology

Yeah I understand what you mean, but it's also important to consider that parents or guardians may set up accounts for their children, and providing the ability to change passwords can give them a sense of ownership and security over their accounts.

**Comment by mehreeee** [comment]
Hi @srish @coderatomy Here is a mockup of what I am proposing.

I've incorporated a settings button on the side panel to address this issue. When clicked, it unveils a panel allowing users to monitor the number of devices logged into the same account. For instance, if a child has multiple iPads or tablets, parents can easily keep tabs. Additionally, if there's a need to change the password, or update a phone number or email, this feature handles it seamlessly. Keep in mind, this is just a mockup, so other elements can be included or removed as needed.
A feedback would be appreciated. Thank you.
![Sign Up option3](https://github.com/unstructuredstudio/zubhub/assets/105458062/26548a5c-fe97-445f-9e2a-d36723c7280e)
![Sign Up option](https://github.com/unstructuredstudio/zubhub/assets/105458062/d6dc6746-0a0a-4037-bda8-0c5cec7724e3)

**Comment by coderatomy** [comment]
Yeah! This looks good. Nice work @mehreeee. Just one thing. Will this mean that will this mean that editting profile will be moved to here?

**Comment by mehreeee** [comment]
> Yeah! This looks good. Nice work @mehreeee. Just one thing. Will this mean that will this mean that editting profile will be moved to here?

Yeah that is what I’m suggesting. Since we’re using Reddit’s model that allows users to remain anonymous to an extent, I would have suggested that if we click on the profile avatar, Users should only be able to change their profile picture.

**Comment by srish** [comment]
Yes, https://github.com/unstructuredstudio/zubhub/issues/904#issuecomment-1762934593 makes sense. Some minor suggestions: Instead of "new password" say "change password". Beneath current password add a text "Forgot Password?" and link it to the reset password route.

**Comment by coderatomy** [comment]
Nice. Taking over

**Comment by mehreeee** [comment]
Hi @srish @coderatomy this is what I came up with. The forgot password text is a clickable link to reset the password
I also added another field for users to retype new passwords. This is good UX because it helps confirm that the user intended to choose a specific password. This prevents accidental typos or mistakes that could result in a forgotten password later. The act of typing the password twice reinforces the memory of the new password. This can help users remember their password more easily, especially if it's a complex one.

The "Retype New Password" field should pop up when the "Change Password" field is selected.


![Edit](https://github.com/unstructuredstudio/zubhub/assets/105458062/01fb3c65-47c7-4a4c-a995-672a05e4ff3b)
![Edit2](https://github.com/unstructuredstudio/zubhub/assets/105458062/e0af08cf-d2ac-47da-a3be-e6bdbd16f4dc)

**Comment by Boluwatifey1** [comment]
> Hi @srish @coderatomy this is what I came up with. The forgot password text is a clickable link to reset the password I also added another field for users to retype new passwords. This is good UX because it helps confirm that the user intended to choose a specific password. This prevents accidental typos or mistakes that could result in a forgotten password later. The act of typing the password twice reinforces the memory of the new password. This can help users remember their password more easily, especially if it's a complex one.
> 
> ![Edit](https://user-images.githubusercontent.com/105458062/275736735-4b3f336b-50c7-434f-986d-578cf5356230.png) ![Edit2](https://user-images.githubusercontent.com/105458062/275736769-393c141b-c2b6-4676-9356-93b1906c192a.png)

this is a nice design, i like the fact that user can retype their password for confirmation

**Comment by mehreeee** [comment]
> > Hi @srish @coderatomy this is what I came up with. The forgot password text is a clickable link to reset the password I also added another field for users to retype new passwords. This is good UX because it helps confirm that the user intended to choose a specific password. This prevents accidental typos or mistakes that could result in a forgotten password later. The act of typing the password twice reinforces the memory of the new password. This can help users remember their password more easily, especially if it's a complex one.

> > ![Edit](https://user-images.githubusercontent.com/105458062/275736735-4b3f336b-50c7-434f-986d-578cf5356230.png) ![Edit2](https://user-images.githubusercontent.com/105458062/275736769-393c141b-c2b6-4676-9356-93b1906c192a.png)
> 
> this is a nice design, i like the fact that user can retype their password for confirmation

Thank you @Boluwatifey1

**Comment by Boluwatifey1** [comment]
I and @julietmadu made a similar design for issue #837 but it was closed because they did not want user to be able to edit their profile picture ,we also made a change password field but the retype password field is actually a better idea

**Comment by yokwejuste** [comment]
> Hi @mehreeee i discovered that too and i made a design for it but it was closed by @srish
> 
> i added a change icon field beside the old password
> 
> ![add Profile picture (6)](https://user-images.githubusercontent.com/127180289/275237004-044b9539-f574-4973-a895-014846be4941.png)

I think we need a separate edit for the password not mixed up with other updates.

**Comment by yokwejuste** [comment]
> Hi @srish @coderatomy Here is a mockup of what I am proposing.
> 
> I've incorporated a settings button on the side panel to address this issue. When clicked, it unveils a panel allowing users to monitor the number of devices logged into the same account. For instance, if a child has multiple iPads or tablets, parents can easily keep tabs. Additionally, if there's a need to change the password, or update a phone number or email, this feature handles it seamlessly. Keep in mind, this is just a mockup, so other elements can be included or removed as needed. A feedback would be appreciated. Thank you. ![Sign Up option3](https://user-images.githubusercontent.com/105458062/275430584-26548a5c-fe97-445f-9e2a-d36723c7280e.png) ![Sign Up option](https://user-images.githubusercontent.com/105458062/275435823-d6dc6746-0a0a-4037-bda8-0c5cec7724e3.png)

Great job @mehreeee, 

I love it, but some questions:
- Why the 2FA?
- The connected are the devices we signed in to? I don't think this is a priority for now, since implementing this will be a little complex and we will need an email service which we don't yet have.
- I guess you mean change email or phone number, focusing on the UX the change should be one after another

**Comment by mehreeee** [comment]
> > > Hi @srish @coderatomy this is what I came up with. The forgot password text is a clickable link to reset the password I also added another field for users to retype new passwords. This is good UX because it helps confirm that the user intended to choose a specific password. This prevents accidental typos or mistakes that could result in a forgotten password later. The act of typing the password twice reinforces the memory of the new password. This can help users remember their password more easily, especially if it's a complex one.
> 
> > > ![Edit](https://user-images.githubusercontent.com/105458062/275736735-4b3f336b-50c7-434f-986d-578cf5356230.png) ![Edit2](https://user-images.githubusercontent.com/105458062/275736769-393c141b-c2b6-4676-9356-93b1906c192a.png)
> > 
> > 
> > this is a nice design, i like the fact that user can retype their password for confirmation
> 
> Thank you @Boluwatifey1

Hi @yokwejuste I made changes to the design and implemented what @srish said here https://github.com/unstructuredstudio/zubhub/issues/904#issuecomment-1765470583

**Comment by mehreeee** [comment]
> > Hi @srish @coderatomy Here is a mockup of what I am proposing.
> > I've incorporated a settings button on the side panel to address this issue. When clicked, it unveils a panel allowing users to monitor the number of devices logged into the same account. For instance, if a child has multiple iPads or tablets, parents can easily keep tabs. Additionally, if there's a need to change the password, or update a phone number or email, this feature handles it seamlessly. Keep in mind, this is just a mockup, so other elements can be included or removed as needed. A feedback would be appreciated. Thank you. ![Sign Up option3](https://user-images.githubusercontent.com/105458062/275430584-26548a5c-fe97-445f-9e2a-d36723c7280e.png) ![Sign Up option](https://user-images.githubusercontent.com/105458062/275435823-d6dc6746-0a0a-4037-bda8-0c5cec7724e3.png)
> 
> Great job @mehreeee,
> 
> I love it, but some questions:
> 
> * Why the 2FA?
> * The connected are the devices we signed in to? I don't think this is a priority for now, since implementing this will be a little complex and we will need an email service which we don't yet have.
> * I guess you mean change email or phone number, focusing on the UX the change should be one after another

I understand the complexity. They were just mockups of what I felt the UX should be. The main idea was a separate page for changing passwords, emails, and phone numbers. I guess they can fit on the same page for now and focus on the more important things.

**Comment by yokwejuste** [comment]
> Yes, [#904 (comment)](https://github.com/unstructuredstudio/zubhub/issues/904#issuecomment-1762934593) makes sense. Some minor suggestions: Instead of "new password" say "change password". Beneath current password add a text "Forgot Password?" and link it to the reset password route.

@srish @tuxology I'm suggesting we should go in for a page for **Password change** and one for **user updates**. This will also be different endpoints on the backend for updates and password change.
![image](https://github.com/unstructuredstudio/zubhub/assets/71908316/51f98404-5e83-4758-aa8c-4255f93d12f3)

**Comment by mehreeee** [comment]
> > Yes, [#904 (comment)](https://github.com/unstructuredstudio/zubhub/issues/904#issuecomment-1762934593) makes sense. Some minor suggestions: Instead of "new password" say "change password". Beneath current password add a text "Forgot Password?" and link it to the reset password route.
> 
> @srish @tuxology I'm suggesting we should go in for a page for **Password change** and one for **user updates**. This will also be different endpoints on the backend for updates and password change. ![image](https://user-images.githubusercontent.com/71908316/275836994-51f98404-5e83-4758-aa8c-4255f93d12f3.png)

I agree. This is the point I've been trying to explain. We can take out the 2FA, Connected devices and email setups for now.

**Comment by DonPresh** [comment]
> > > > Hi @srish @coderatomy this is what I came up with. The forgot password text is a clickable link to reset the password I also added another field for users to retype new passwords. This is good UX because it helps confirm that the user intended to choose a specific password. This prevents accidental typos or mistakes that could result in a forgotten password later. The act of typing the password twice reinforces the memory of the new password. This can help users remember their password more easily, especially if it's a complex one.
> > 
> > 
> > > > ![Edit](https://user-images.githubusercontent.com/105458062/275736735-4b3f336b-50c7-434f-986d-578cf5356230.png) ![Edit2](https://user-images.githubusercontent.com/105458062/275736769-393c141b-c2b6-4676-9356-93b1906c192a.png)
> > > 
> > > 
> > > this is a nice design, i like the fact that user can retype their password for confirmation
> > 
> > 
> > Thank you @Boluwatifey1
> 
> Hi @yokwejuste I made changes to the design and implemented what @srish said here [#904 (comment)](https://github.com/unstructuredstudio/zubhub/issues/904#issuecomment-1765470583)

@mehreeee I like this.  Just a few corrections I think you should make:
-The part where you said "Tell us something fun" should be inside the text box (I'm imagining that description is for the bio textbox)
-"Forgot password" should be directed under "current password" text box so it's clear to users that, the action required is directed linked to "current password"
-Just a UI observation, Make your spacing consistent.

Other than these corrections, love your designs.

**Comment by mehreeee** [comment]
> > > > > Hi @srish @coderatomy this is what I came up with. The forgot password text is a clickable link to reset the password I also added another field for users to retype new passwords. This is good UX because it helps confirm that the user intended to choose a specific password. This prevents accidental typos or mistakes that could result in a forgotten password later. The act of typing the password twice reinforces the memory of the new password. This can help users remember their password more easily, especially if it's a complex one.
> > > 
> > > 
> > > > > ![Edit](https://user-images.githubusercontent.com/105458062/275736735-4b3f336b-50c7-434f-986d-578cf5356230.png) ![Edit2](https://user-images.githubusercontent.com/105458062/275736769-393c141b-c2b6-4676-9356-93b1906c192a.png)
> > > > 
> > > > 
> > > > this is a nice design, i like the fact that user can retype their password for confirmation
> > > 
> > > 
> > > Thank you @Boluwatifey1
> > 
> > 
> > Hi @yokwejuste I made changes to the design and implemented what @srish said here [#904 (comment)](https://github.com/unstructuredstudio/zubhub/issues/904#issuecomment-1765470583)
> 
> @mehreeee I like this. Just a few corrections I think you should make: -The part where you said "Tell us something fun" should be inside the text box (I'm imagining that description is for the bio textbox) -"Forgot password" should be directed under "current password" text box so it's clear to users that, the action required is directed linked to "current password" -Just a UI observation, Make your spacing consistent.
> 
> Other than these corrections, love your designs.

Thank you @DonPresh I have effected the corrections. 

Quick question does the "forgot password" text look better directly under the current password text box or at the end of the text box? 
![Edit3](https://github.com/unstructuredstudio/zubhub/assets/105458062/6434ad18-5fd2-4d8a-8d1f-03de4e219977)

**Comment by DonPresh** [comment]
@mehreeee end of text box

**Comment by coderatomy** [comment]
Hey @srish @yokwejuste @DonPresh @mehreeee. I still have one doubt here that maybe you can help me solve.
I saw we have support for changing password from `/password-reset` `password-reset-confirm`. Does this mean we are completely shifting from that? If yes, then what will become of it?

**Comment by mehreeee** [comment]
> Hey @srish @yokwejuste @DonPresh @mehreeee. I still have one doubt here that maybe you can help me solve. I saw we have support for changing password from `/password-reset` `password-reset-confirm`. Does this mean we are completely shifting from that? If not, then what will become of it?

I'm not sure I understand your question.

**Comment by yokwejuste** [comment]
> Hey @srish @yokwejuste @DonPresh @mehreeee. I still have one doubt here that maybe you can help me solve. I saw we have support for changing password from `/password-reset` `password-reset-confirm`. Does this mean we are completely shifting from that? If yes, then what will become of it?

Hey @coderatomy  is that on the frontend or backend ?

**Comment by srish** [comment]
Let's go with the design implemented by @mehreeee and @Boluwatifey1 one in which there is an option to retype password too https://github.com/unstructuredstudio/zubhub/issues/904#issuecomment-1765843102

**Comment by coderatomy** [comment]
Hey @mehreeee. Can you share the figma link for your mock up

**Comment by mehreeee** [comment]
> Hey @mehreeee. Can you share the figma link for your mock up

Sure here's the Figma link 
https://www.figma.com/file/55Wfr1bOEevZHlAu1DLzkv/Edit-profile-(Copy)?type=design&node-id=0%3A1&mode=design&t=xnFb5TurQjULxKRO-1

**Comment by JulietMadu** [comment]
@coderatomy does it automatically update for mobile too? If no , I can work on the mobile version @srish

**Comment by coderatomy** [comment]
You can just take over with this @JulietMadu, if that's fine with you

**Comment by JulietMadu** [comment]
> > Hey @mehreeee. Can you share the figma link for your mock up
> 
> Sure here's the Figma link https://www.figma.com/file/55Wfr1bOEevZHlAu1DLzkv/Edit-profile-(Copy)?type=design&node-id=0%3A1&mode=design&t=xnFb5TurQjULxKRO-1

@mehreeee can you place the "forgot password" text underneath the current password box . I feel it will be better that way. 
Also when a user click on the edit profile and it takes them to this page, I suggest that the both the current password and new password box should be empty... I created the mobile version and made some dew changes @srish 
![Screenshot_20231020-135541_Chrome](https://github.com/unstructuredstudio/zubhub/assets/128796937/570444ef-b8e2-46d0-a2ba-901168a63cdd)

**Comment by mehreeee** [comment]
> > > Hey @mehreeee. Can you share the figma link for your mock up
> > 
> > 
> > Sure here's the Figma link https://www.figma.com/file/55Wfr1bOEevZHlAu1DLzkv/Edit-profile-(Copy)?type=design&node-id=0%3A1&mode=design&t=xnFb5TurQjULxKRO-1
> 
> @mehreeee can you place the "forgot password" text underneath the current password box . I feel it will be better that way. Also when a user click on the edit profile and it takes them to this page, I suggest that the both the current password and new password box should be empty... I created the mobile version and made some dew changes @srish ![Screenshot_20231020-135541_Chrome](https://user-images.githubusercontent.com/128796937/276963803-570444ef-b8e2-46d0-a2ba-901168a63cdd.jpg)

Hi @JulietMadu the design @srish chose didn’t have the forgot password link under the current text box. Yeah the current and new password text boxes will be empty. Those were just placeholders.
